backend/src/core/middlewares/campaign.middleware.ts

```diff
@@ -151,9 +153,39 @@ const isCampaignRedacted = async (
   }
 }
 
+/**
+ * Validate url origin is from Vault and check url expiry
+ */
+const isValidVaultUrl = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Response | void => {
+  try {
+    const { url } = req.body
+    const vaultUrl = new URL(url)
+    const expiry = vaultUrl.searchParams.get('Expires')
+    const datasetName = vaultUrl.searchParams.get('datasetname')
+
+    if (vaultUrl.hostname !== VAULT_DOMAIN || !expiry || !datasetName) {
+      return res.status(400).json({ message: 'This is not a valid Vault url' })
+    }
+
+    const expiryTimestamp = parseInt(expiry) * 1000
+    if (expiryTimestamp < Date.now()) {
+      return res.status(400).json({ message: 'This is url has expired' })
```

Comment:
```suggestion
      return res.status(400).json({ message: 'This url has expired' })
```
nit: Minor grammatical error